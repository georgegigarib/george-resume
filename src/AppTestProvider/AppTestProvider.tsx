import React from 'react'
import { I18nextProvider } from 'react-i18next'
import i18n from '@/locales/i18n.config'
import store from '@/store'
import { Provider } from 'react-redux'
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom'
import { vi } from 'vitest'

vi.mock('swiper/react', () => ({
  Swiper: ({
    children,
    onSlideChange,
  }: {
    children: React.ReactNode
    onSlideChange?: (params: { realIndex: number }) => void
  }) => {
    onSlideChange?.({ realIndex: 0 })
    return <div data-testid="swiper">{children}</div>
  },
  SwiperSlide: ({ children }: { children: React.ReactNode }) => <div data-testid="swiper-slide">{children}</div>,
}))

const AppTestProvider = ({ children }: { children: React.ReactNode }) => {
  return (
    <I18nextProvider i18n={i18n}>
      <Provider store={store}>
        <Router>
          <Routes>
            <Route path="/" element={children} />
          </Routes>
        </Router>
      </Provider>
    </I18nextProvider>
  )
}

export default AppTestProvider
