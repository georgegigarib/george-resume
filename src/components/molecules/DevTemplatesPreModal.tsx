import CodeTemplate from '@/assets/icons/code-template.webp'
import ImageWithLoader from '@/components/atoms/ImageLoader.tsx'
import { useTranslation } from 'react-i18next'
import { Tooltip } from '@mui/material'
import AppsIcon from '@mui/icons-material/Apps'
import { motion } from 'framer-motion'
import React from 'react'

export default function DevTemplatesPreModal(): React.ReactElement {
  const { t } = useTranslation()

  return (
    <div className="relative h-full w-full py-3 px-2 rounded-xl cursor-pointer font-semibold ">
      <Tooltip arrow title={t('dev.templates.tooltip.showAllTemplates')}>
        <motion.div
          className="absolute top-2 right-2 block hover:bg-white hover:bg-opacity-25 px-1 py-1 rounded-full text-black dark:text-white"
          whileHover={{ scale: 1.07 }}
        >
          <AppsIcon fontSize="medium" />
        </motion.div>
      </Tooltip>

      <h2 className="text-black dark:text-white">{t('dev.templates.title')}</h2>

      <motion.div
        initial={{ z: 180 }}
        animate={{ rotate: [0, 70, 0] }}
        whileHover={{ rotate: 720, scale: 1.2 }}
        className="absolute -bottom-[25.7rem] right-2 text-xs hidden md:block"
        transition={{
          duration: 0.6,
          repeat: Infinity,
          ease: 'linear',
          repeatDelay: 2,
        }}
      >
        <div className="w-10">
          <ImageWithLoader imagePath={CodeTemplate} />
        </div>
      </motion.div>
    </div>
  )
}
