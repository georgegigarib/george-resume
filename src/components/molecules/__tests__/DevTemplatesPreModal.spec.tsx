import { render } from '@testing-library/react'
import { describe, expect, it, vi, beforeEach } from 'vitest'
import DevTemplatesPreModal from '@/components/molecules/DevTemplatesPreModal'
import AppTestProvider from '@/AppTestProvider/AppTestProvider'

describe('DevTemplatesPreModal component', () => {
  beforeEach(() => {
    vi.clearAllMocks()
  })

  const renderComponent = () =>
    render(
      <AppTestProvider>
        <DevTemplatesPreModal />
      </AppTestProvider>
    )

  it('matches the snapshot', () => {
    const { asFragment } = renderComponent()

    expect(asFragment()).toMatchSnapshot()
  })
})
