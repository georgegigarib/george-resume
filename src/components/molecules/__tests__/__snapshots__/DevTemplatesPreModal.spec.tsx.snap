// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DevTemplatesPreModal component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="relative h-full w-full py-3 px-2 rounded-xl cursor-pointer font-semibold "
  >
    <div
      aria-label="Show all templates"
      class="absolute top-2 right-2 block hover:bg-white hover:bg-opacity-25 px-1 py-1 rounded-full text-black dark:text-white"
      data-mui-internal-clone-element="true"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-20bmp1-MuiSvgIcon-root"
        data-testid="AppsIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M4 8h4V4H4zm6 12h4v-4h-4zm-6 0h4v-4H4zm0-6h4v-4H4zm6 0h4v-4h-4zm6-10v4h4V4zm-6 4h4V4h-4zm6 6h4v-4h-4zm0 6h4v-4h-4z"
        />
      </svg>
    </div>
    <h2
      class="text-black dark:text-white"
    >
      Use my dev templates!
    </h2>
    <div
      class="absolute -bottom-[25.7rem] right-2 text-xs hidden md:block"
      style="will-change: transform; transform: translateZ(180px);"
    >
      <div
        class="w-10"
      >
        <div
          class="absolute inset-0 flex items-center justify-center"
        >
          <span
            class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1yonwsg-MuiCircularProgress-root"
            role="progressbar"
            style="width: 40px; height: 40px;"
          >
            <svg
              class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-1a8epmm-MuiCircularProgress-circle"
                cx="44"
                cy="44"
                fill="none"
                r="20.2"
                stroke-width="3.6"
              />
            </svg>
          </span>
        </div>
        <img
          class="undefined opacity-0"
          src="/src/assets/icons/code-template.webp"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
